src/skin/qml/qmleffectmanifestparametersmodel.cpp

```diff
@@ -0,0 +1,73 @@
+#include "skin/qml/qmleffectmanifestparametersmodel.h"
+
+#include <QModelIndex>
+
+#include "effects/effectmanifest.h"
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+namespace {
+const QHash<int, QByteArray> kRoleNames = {
+        {QmlEffectManifestParametersModel::IdRole, "id"},
+        {QmlEffectManifestParametersModel::NameRole, "name"},
+        {QmlEffectManifestParametersModel::ShortNameRole, "shortName"},
+        {QmlEffectManifestParametersModel::DescriptionRole, "description"},
+        {QmlEffectManifestParametersModel::ControlHintRole, "controlHint"},
+};
+}
+
+QmlEffectManifestParametersModel::QmlEffectManifestParametersModel(
+        EffectManifestPointer pEffectManifest,
+        QObject* parent)
+        : QAbstractListModel(parent), m_pEffectManifest(pEffectManifest) {
+}
+
+QVariant QmlEffectManifestParametersModel::data(const QModelIndex& index, int role) const {
+    const QList<EffectManifestParameterPointer>& parameters = m_pEffectManifest->parameters();
+    if (index.row() >= parameters.size()) {
+        return QVariant();
+    }
+
+    const EffectManifestParameterPointer pParameter = parameters.at(index.row());
+    switch (role) {
+    case QmlEffectManifestParametersModel::IdRole:
+        return pParameter->id();
+    case QmlEffectManifestParametersModel::NameRole:
+        return pParameter->name();
+    case QmlEffectManifestParametersModel::ShortNameRole:
+        return pParameter->shortName();
+    case QmlEffectManifestParametersModel::DescriptionRole:
+        return pParameter->description();
+    case QmlEffectManifestParametersModel::ControlHintRole:
+        return static_cast<int>(pParameter->controlHint());
+    default:
+        return QVariant();
+    }
+}
+
+int QmlEffectManifestParametersModel::rowCount(const QModelIndex& parent) const {
+    if (parent.isValid()) {
+        return 0;
+    }
```

Comment:
```suggestion
    if (!parent.isValid()) {
        return 0;
    }
```
not sure how the QModelIndex API is used, but returning early when the parent is valid seems strange.

Reply:
This model is a *list*. This means it has N rows. If *parent* is valid, this means that we're not requesting the rowCount for the whole list, but for an individual *item*. If this was a tree model (like the library sidebar), then each item might also have children. This is not the case here. Therefore, we always return zero, because no item has children.

Reply:
I understand now. Thanks for the explanation. I guess I should read the Qt docs first without asking questions in code review.